ClassDataSource in partial classes "breaks" compilation when compiling.  warning CS8785

Summary

When using partial classes together with ClassDataSource, the source generators (DataSourceHelpersGenerator and PropertyInjectionSourceGenerator) fail due to duplicate `"hintName"` values.

Minimal Repro
```
public partial class Tests
{
    [Test]
    public void Basic()
    {
        Console.WriteLine("This is a basic test");
    }
}

public partial class Tests
{
    [ClassDataSource<DataClass>(Shared = SharedType.PerTestSession)]
    public required DataClass DataClass { get; init; }
}
```
Actual Behaviour

Compilation produces the following warnings/errors:

`CSC : warning CS8785: Generator 'DataSourceHelpersGenerator' failed to generate source. 
It will not contribute to the output and compilation errors may occur as a result. 
Exception was of type 'ArgumentException' with message 
'The hintName 'PartialTest_Tests_DataSourceHelper.g.cs' of the added source file 
must be unique within a generator. (Parameter 'hintName')'.

CSC : warning CS8785: Generator 'PropertyInjectionSourceGenerator' failed to generate source. 
It will not contribute to the output and compilation errors may occur as a result. 
Exception was of type 'ArgumentException' with message 
'The "hintName" 'PartialTest_Tests_PropertyInjection.g.cs' of the added source file 
must be unique within a generator. (Parameter 'hintName')'.`

Expected Behaviour

Partial classes should work seamlessly with ClassDataSource without causing duplicate "hintName" collisions in the generated files.
This is probably du to the source generator assigning hintName as some internal thing or not gripping the name of the violator.

Notes

Tested with partial class split across two files.

Works fine if everything is placed in a single class declaration.

Unsure if this is intended or a bug in the generators’ handling of partials.

Tested on the newest version 0.61.39

--- a/TUnit.Core.SourceGenerator/Generators/DataSourceHelpersGenerator.cs
+++ b/TUnit.Core.SourceGenerator/Generators/DataSourceHelpersGenerator.cs
@@ -17,9 +17,11 @@ public class DataSourceHelpersGenerator : IIncrementalGenerator
             .CreateSyntaxProvider(
                 predicate: static (s, _) => s is ClassDeclarationSyntax,
                 transform: static (ctx, _) => GetTypeWithDataSourceProperties(ctx))
-            .Where(static t => t is not null);
+            .Where(static t => t is not null)
+            .Collect()
+            .SelectMany((types, _) => types.DistinctBy(t => t!.Value.TypeSymbol, SymbolEqualityComparer.Default));
 
-        // Generate individual files for each type instead of collecting them all
+        // Generate individual files for each unique type
         context.RegisterSourceOutput(typesWithDataSourceProperties, (spc, type) => { if (type != null) GenerateIndividualDataSourceHelper(spc, type); });
     }
 

--- a/TUnit.Core.SourceGenerator/Generators/PropertyInjectionSourceGenerator.cs
+++ b/TUnit.Core.SourceGenerator/Generators/PropertyInjectionSourceGenerator.cs
@@ -16,9 +16,11 @@ public sealed class PropertyInjectionSourceGenerator : IIncrementalGenerator
                 predicate: (node, _) => IsClassWithDataSourceProperties(node),
                 transform: (ctx, _) => GetClassWithDataSourceProperties(ctx))
             .Where(x => x != null)
-            .Select((x, _) => x!);
+            .Select((x, _) => x!)
+            .Collect()
+            .SelectMany((classes, _) => classes.DistinctBy(c => c.ClassSymbol, SymbolEqualityComparer.Default));
 
-        // Generate individual files for each class instead of collecting them all
+        // Generate individual files for each unique class
         context.RegisterSourceOutput(classesWithPropertyInjection, GenerateIndividualPropertyInjectionSource);
     }
 
@@ -87,6 +89,12 @@ public sealed class PropertyInjectionSourceGenerator : IIncrementalGenerator
                     break; // Only one data source per property
                 }
             }
+        }
+
+        // Only return if there are actually properties with data sources
+        if (propertiesWithDataSources.Count == 0)
+        {
+            return null;
         }
 
         return new ClassWithDataSourceProperties

--- a/TUnit.TestProject/Bugs/3231/DataClass.cs
+++ b/TUnit.TestProject/Bugs/3231/DataClass.cs
@@ -1,0 +1,6 @@
+namespace TUnit.TestProject.Bugs._3231;
+
+public class DataClass
+{
+    public string Value { get; set; } = "Test Data";
+}

--- a/TUnit.TestProject/Bugs/3231/TestsPart1.cs
+++ b/TUnit.TestProject/Bugs/3231/TestsPart1.cs
@@ -1,0 +1,13 @@
+using TUnit.TestProject.Attributes;
+
+namespace TUnit.TestProject.Bugs._3231;
+
+[EngineTest(ExpectedResult.Pass)]
+public partial class Tests
+{
+    [Test]
+    public void Basic()
+    {
+        Console.WriteLine($"This is a basic test with data: {DataClass.Value}");
+    }
+}

--- a/TUnit.TestProject/Bugs/3231/TestsPart2.cs
+++ b/TUnit.TestProject/Bugs/3231/TestsPart2.cs
@@ -1,0 +1,7 @@
+namespace TUnit.TestProject.Bugs._3231;
+
+public partial class Tests
+{
+    [ClassDataSource<DataClass>(Shared = SharedType.PerTestSession)]
+    public required DataClass DataClass { get; init; }
+}

